coingecko/coingecko-fetcher.ts

```diff
@@ -1,123 +0,0 @@
-import { kv } from "@vercel/kv";
-import moment from "moment";
-import { userKey } from "./constants";
-import { coingeckoAPIErrorResponse, invalidResponseTypesError } from "./errors";
-import { PriceDataResponse } from "./interfaces";
-import { coinList } from "./supported-coins";
-import {
-  CoingeckoResponse,
-  getCoingeckoRangeURL,
-  getDayId,
-  getTimestampFromDayId,
-  KVDataToPrice,
-  parsePriceResponse,
-} from "./utils";
-
-const fetchData = async (symbol: string, start: number, finish: number) => {
-  const id = coinList[symbol];
-  const url = getCoingeckoRangeURL(id, start, finish);
-  const res = await fetch(
-    `https://api.scraperapi.com/?api_key=${
-      process.env.SCRAPER_API_KEY
-    }&url=${encodeURIComponent(url)}`
-  );
-
-  if (res.status !== 200) {
-    throw coingeckoAPIErrorResponse(res);
-  }
-  const rawResponse = await res.json();
-
-  const parsed = CoingeckoResponse.safeParse(rawResponse);
-
-  if (parsed.success === false) {
-    console.error(parsed.error.issues);
-    throw invalidResponseTypesError;
-  }
-
-  return parsePriceResponse(parsed.data.prices);
-};
-
-export const fetchCoingeckoPrices = async (
-  assets: string[],
-  start: number,
-  days: number
-): Promise<PriceDataResponse> => {
-  const startTimestamp = moment(start * 1000)
-    .utc()
-    .startOf("day")
-    .unix();
-
-  const finish = moment(startTimestamp * 1000)
-    .utc()
-    .add(days, "days")
-    .startOf("day")
-    .unix();
-
-  const currentTimestamp = moment().unix();
-  const finishTimestamp =
-    finish > currentTimestamp
-      ? moment(currentTimestamp * 1000)
-          .utc()
-          .startOf("day")
-          .unix()
-      : finish;
```

Comment:
if you like moment.js THAT much :shipit: , than use its full power, like https://www.geeksforgeeks.org/moment-js-moment-min-method/

and, there is no need for the intermediate currentTimestamp variable – you're getting it on the fly